agent/config/builder.go

```diff
@@ -1089,6 +1090,10 @@ func (b *Builder) Validate(rt RuntimeConfig) error {
 		}
 	}
 
+	//MinQuorum is set to BootstrarpExpect if available
+	if rt.BootstrapExpect != 0 {
```

Comment:
I'm sure there will be a time someone wants to set both BootstrapExpect and MinQuorum. Do we want them to be able to be different, or should it override MinQuorum if BootstrapExpect is set?
Open to discussion here.

Reply:
I think we should allow bootstrap expect to be lower than MinQuorum but not the other way around

Reply:
Hmm - okay, I can totally do that. Do you have an example of how they would be set that way/ why someone would use them that way? I'm curious because I cannot think of why someone would want MinQuorum and BootstrapExpect to be different, but I know there **has** to be users who would want it.

Reply:
So thinking a little more my original answer I would alter to say that we shouldn't enforce any particular constraints the relationship of minquorum and bootstrap expect.

So first, the autopilot settings in the config are only ever used if no autopilot config exists (so only when bootstrapping a new datacenter). Otherwise you must manage it from the API/CLI.

When managing from the API/CLI we don't even have this question because the bootstrap expect value isn't persisted in the raft autopilot config. It could be that you originally wanted to use 5 servers but found you only needed 3 or maybe you are going the other way. Regardless whatever value you set in this fashion should be taken (unless its a value that is just bad like 0 or something which never makes sense).

So given that the only time when there may be a correlation between the bootstrap expect value and the min quorum size is during dc bootstrapping for the first time when deciding on the autopilot config to set in raft.

If MinQuorum is set we should use it as the user "knows what they are doing".

If its less than bootstrap expect maybe they want to wait for all 5 servers to come online before bootstrapping everything for the first time but once operational they want more availability so they will allow the min quorum to be 3.

On the other side, maybe they want to get to bootstrapping things and need to wait for 2 or 3 servers but eventually will have 5 and want that to be the minimum that autopilot will reduce things to.

Both cases seem plausible to me.

Reply:
Right, I see. So would you rather remove the relationship between the two or keep it as is (if MinQuorum is not set, take on BootstrapExpect value) and also create a validator for 0 value inputs (ex: if users inputs 0, throw an err?)

Reply:
I think leaving as is would be good. If MinQuorum isn't set then assume it should be the bootstrap expect value.

Does 0 indicate that the min quorum size calculations should be disabled?

Reply:
I guess that would make sense, should people are the option to set MinQuorum to 0 and then set BootstrapExpect to 5 (or whatever)

Reply:
I think so. If you want todays behavior then you need to disable the MinQuorum calculations but you may also want to still do bootstrap expect stuff.

Reply:
I think this new update is much better. :)